core/app.go

```diff
@@ -373,3 +376,37 @@ func (a *App) stopCoprocesses() {
 		coprocess.Stop()
 	}
 }
+
+// getStatusHandler generates HTTP response for Telemetry status control plane
+// endpoint.
+func (a *App) getStatusHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test for GET requests
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+
+	servicesJSON, err := json.Marshal(a.Services)
+	if err != nil {
+		log.Errorf("Error generating services for '/status': %v", err)
+	}
+	log.Debugf("Marshaled services: %v", string(servicesJSON))
+
+	// w.Write([]byte(`{ "message": true }`))
+	w.Write(servicesJSON)
+}
+
+// postEnvHandler generates HTTP response for updating environment variables of
+// future CP managed processes.
+func (a *App) postEnvHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test HTTP method
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write([]byte(`{ "message": true }`))
+}
+
+func (a *App) handleServer() *control.Server {
+	s := a.Server
+	s.Mux.HandleFunc("/status", a.getStatusHandler)
+	s.Mux.HandleFunc("/v3/env", a.postEnvHandler)
+	s.Serve()
```

Comment:
I'd like to minimize the surface area of `App` as much as possible. As a bit of history here the original interface of both `core/App` and `config/Config` was a lot smaller but as we added features we didn't do an awesome job of refactoring the internals to match the greater complexity. 

In https://github.com/joyent/containerpilot/pull/290 I'm going work to make the various packages pass messages on an event bus. So the more strongly you can separate `core/App` from the server the better this will all come together when it's time to integrate them.